graylog2-web-interface/src/views/components/searchbar/QueryInput.tsx

```diff
@@ -156,14 +163,15 @@ const QueryInput = ({
 }: Props) => {
   const { data: queryFields } = useFieldTypes(streams, isNoTimeRangeOverride(timeRange) ? DEFAULT_TIMERANGE : timeRange);
   const { data: allFields } = useFieldTypes([], DEFAULT_TIMERANGE);
+  const isInitialTokenizerUpdate = useRef(true);
```

Comment:
If you don't need this ref in your JSX code you can do I other way.

```
const [isInitialTokenizerUpdate, setIsInitialTokenizerUpdate ]= useState(true);
...
const onLoadEditor = useCallback((editor: Editor) => _onLoadEditor(editor, setIsInitialTokenizerUpdate), [ setIsInitialTokenizerUpdate ]);
... 

const _onLoadEditor = (editor, setIsInitialTokenizerUpdate: React.MutableRefObject<boolean>) => {
  if (editor) {
    editor.commands.removeCommands(['find', 'indent', 'outdent']);

    editor.session.on('tokenizerUpdate', (input, { bgTokenizer: { currentLine, lines } }) => {
      setIsInitialTokenizerUpdate( isInitialTokenizerUpdate => {
            if (isInitialTokenizerUpdate.current) {
               return true
            }
      })
        editor.completers.forEach((completer) => {
          if (completer?.shouldShowCompletions(currentLine, lines)) {
            editor.execCommand('startAutocomplete');
          }
        });    
        return isInitialTokenizerUpdate
        });

    editor.renderer.setScrollMargin(6, 5);
    editor.renderer.setPadding(12);
  }
};
```

Reply:
What would be the benefit? I prefer using a ref in this case to avoid triggering a state update.